components/feature/session/src/main/java/mozilla/components/feature/session/DefaultSessionStorage.kt

```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session
+
+import android.content.Context
+import mozilla.components.browser.session.Session
+import mozilla.components.concept.engine.Engine
+import mozilla.components.concept.engine.EngineSession
+import org.json.JSONObject
+import java.io.FileNotFoundException
+import java.util.UUID
+
+const val SELECTED_SESSION_KEY = "selectedSession"
+const val SESSION_KEY = "session"
+const val ENGINE_SESSION_KEY = "engineSession"
+
+/**
+ * Default implementation of [SessionStorage] which persists browser and engine
+ * session states to a JSON file.
+ *
+ * The JSON format used for persisting:
+ * {
+ *     "selectedSession": "[session-uuid]",
+ *     "[session-uuid]": {
+ *         "session": {}
+ *         "engineSession": {}
+ *     },
+ *     "[session-uuid]": {
+ *         "session": {}
+ *         "engineSession": {}
+ *     }
+ * }
+ */
+class DefaultSessionStorage(private val context: Context) : SessionStorage {
+    private val defaultFileName = DefaultSessionStorage::class.java.canonicalName.replace(".", "_") + ".json"
```

Comment:
Do we really want to migrate this file whenever we rename or move this class? :)